master/internal/api_experiment_intg_test.go

```diff
@@ -1664,3 +1667,74 @@ func TestExperimentSearchApiFilterParsing(t *testing.T) {
 		require.Equal(t, fmt.Sprintf(`SELECT * WHERE %v`, c[1]), q.String())
 	}
 }
+
+func TestDeleteExperiments(t *testing.T) {
+	var mockRM mocks.ResourceManager
+	// Need _anything_ to error to check the error flow leaves things in DELETE_FAILED and that they are delete-able
+	// still after that.
+	mockRM.On("DeleteJob", mock.Anything).Return(func(sproto.DeleteJob) sproto.DeleteJobResponse {
+		errC := make(chan error, 1)
+		errC <- errors.New("something real bad")
+		return sproto.DeleteJobResponse{Err: errC}
+	}, nil)
+
+	api, curUser, ctx := setupAPITest(t, nil, &mockRM)
+
+	var expIDs []int32
+	for i := 0; i < 3; i++ {
+		exp := createTestExp(t, api, curUser)
+		_, err := db.Bun().NewUpdate().Table("experiments").
+			Set("state = ?", model.CompletedState).
+			Where("id = ?", exp.ID).Exec(ctx)
+		require.NoError(t, err)
+		expIDs = append(expIDs, int32(exp.ID))
+	}
+
+	t.Log("if DeleteExperiment fails, all experiments become DELETE_FAILED")
+	_, err := api.DeleteExperiments(ctx, &apiv1.DeleteExperimentsRequest{ExperimentIds: expIDs})
+	require.NoError(t, err)
+
+	var success bool
+	for i := 0; i < 15; i++ {
+		var inDeleteFailed int
+		for _, expID := range expIDs {
+			e, err := api.GetExperiment(ctx, &apiv1.GetExperimentRequest{ExperimentId: expID})
+			require.NoError(t, err, "expected experiment to go to DELETE_FAILED")
+			if e.Experiment.State != experimentv1.State_STATE_DELETE_FAILED {
```

Comment:
```suggestion
			if e.Experiment.State == experimentv1.State_STATE_DELETE_FAILED {
```
You meant ==?

Reply:
ahh, good catch.

Reply:
lucky me, test still passes just fine with it fixed.